### PR TITLE
Bug/object compare timestamps

### PR DESCRIPTION
**Pull request by mdegerne**
Add test to verify the fix for containers also fixes metadata update propagation for objects.

### CODE REVIEW

File: test/integration/test_migrator.py [hunk 1]
```diff
@@ -716,6 +716,51 @@ def is_where(val):
...
+                raise
+            return False
+
+        conn_remote.put_object(
+            migration['aws_bucket'], key, StringIO.StringIO(content),
+            headers={where_header: 'remote'})
+        conn_local.put_object(
+            migration['container'], key, StringIO.StringIO(content),
+            headers={where_header: 'local'})
+        with self.migrator_running():
```
Comment (timuralp) [inline test/integration/test_migrator.py]: I'm similarly concerned about this test as the container one. Walking through what it does again:
1. create two containers and put two objects, with differing metadata, with the destination object being after the source, so source.ts <= destination.ts.
2. wait 3 seconds to make sure the destination metadata is not overwritten
3. update source metadata and make sure the update propagates
4. update destination metadata and wait 3 seconds to make sure it is not overwritten
5. update source metadata and make sure the update propagates

It seems we test something twice and we are testing metadata propagation for which we already have a test. Similar to the container test, I'd suggest setting up the test as:
1. create two objects as is done currently
2. post on the source object (make sure it's after the destination last-modified date); post on the destination object
3. make sure the source metadata does not overwrite the destination metadata by running the migrator for 3 seconds.

By the way, it'd be great to add a TODO that we should explicitly call the migrator, as the 3 second timeout is hokey.

Comment (mdegerne) [reply test/integration/test_migrator.py]: Fixed, I think.

**Comment by mdegerne** [comment]
rebased

**Comment by mdegerne** [comment]
Fixed test - should behave correctly now.

**Pull request merged**